### PR TITLE
DDF-5502 Upgrading Apache Camel to 2.24.2

### PR DESCRIPTION
**Pull request by aaronilovici**
#### What does this PR do?
This PR updates the version of Apache Camel dependency, which will help reduce the amount of high severity vulnerabilities.

#### Who is reviewing it? 
@mdang8
@rfding
@josephthweatt
@bakejeyner

#### Select relevant component teams: 
@codice/build 
@codice/security

#### Ask 2 committers to review/merge the PR and tag them here.
@brjeter
@oconnormi
@stustison

#### How should this be tested?
This should be non-invasive and does not add any functionality to test. This branch can be checked out, and built using 'mvn clean install' to ensure that it passes all current compilation, unit test, and integration test steps.
#### What are the relevant tickets?
Fixes: #5502

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by aaronilovici** [comment]
build now

**Comment by aaronilovici** [comment]
build now

File: features/camel/src/main/feature/feature.xml [hunk 1]
```diff
@@ -27,7 +27,7 @@
     <!--START Camel Features-->
     <!-- DO NOT EDIT
         All feature definitions come from 3rd party feature file:
-        <repository>mvn:org.apache.camel.karaf/apache-camel/2.22.0/xml/features</repository>
+        <repository>mvn:org.apache.camel.karaf/apache-camel/2.24.2/xml/features</repository>
```
Comment (brjeter) [inline features/camel/src/main/feature/feature.xml]: ✏️ It looks like camel updated the netty version that they use to `4.1.32.Final`, so we should probably update the version here
https://github.com/codice/ddf/blob/a755c443eccb3696ab62b09f3e906c51402b355d/pom.xml#L258

Comment (aaronilovici) [reply features/camel/src/main/feature/feature.xml]: Can do. I had that upgrade on our sec ticket backlog but I can do it here instead 👍

**Comment by aaronilovici** [comment]
build now

**Comment by aaronilovici** [comment]
build now

**Comment by aaronilovici** [comment]
build now

**Comment by oconnormi** [comment]
Build Now

**Comment by aaronilovici** [comment]
build now

**Comment by aaronilovici** [comment]
build now

**Pull request merged**